src/frontend/js/components/DashBoard/router.spec.tsx

```diff
@@ -0,0 +1,115 @@
+import { act } from 'react-dom/test-utils';
+import DashBoardNavigator from './Router';
+import { REACT_QUERY_SETTINGS, RICHIE_USER_TOKEN } from '../../settings';
+import { SessionProvider } from '../../data/SessionProvider';
+import createQueryClient from '../../utils/react-query/createQueryClient';
+import { location } from '../../utils/indirection/window';
+
+jest.mock('utils/indirection/window', () => ({
+  location: {
+    assign: jest.fn(),
+  },
+}));
+
+jest.mock('utils/context', () => ({
+  __esModule: true,
+  default: mockFactories
+    .ContextFactory({
+      authentication: { backend: 'fonzie', endpoint: 'https://demo.endpoint' },
+      joanie_backend: { endpoint: 'https://joanie.endpoint' },
+    })
+    .generate(),
+}));
+
+describe('<DashBoardNavigator />', () => {
+  const Wrapper = ({ initial_entries }: { initial_entries: Array<string> }) => (
+    <QueryClientProvider client={createQueryClient({ persistor: true })}>
+      <IntlProvider locale="en">
+        <SessionProvider>
+          <MemoryRouter initialEntries={initial_entries}>
+            <DashBoardNavigator />
+          </MemoryRouter>
+        </SessionProvider>
+      </IntlProvider>
+    </QueryClientProvider>
+  );
+
+  const initializeUser = (loggedin = true) => {
+    const user = loggedin ? FonzieUserFactory.generate() : null;
+
+    sessionStorage.setItem(
+      REACT_QUERY_SETTINGS.cacheStorage.key,
+      JSON.stringify(
+        PersistedClientFactory({ queries: [QueryStateFactory('user', { data: user })] }),
+      ),
+    );
+
+    if (loggedin) {
+      sessionStorage.setItem(RICHIE_USER_TOKEN, user.access_token);
+    }
+
+    return user;
+  };
+
+  afterEach(() => {
+    sessionStorage.clear();
+  });
+
+  it('shows a navigator', async () => {
+    await act(async () => {
+      render(<Wrapper initial_entries={['/']} />);
+    });
+
+    const paths = ['/courses', '/preferences'];
+    const links = screen.getAllByRole('link');
+    let hrefs: Array<string> = [];
+
+    hrefs = links.reduce((res: Array<string>, item) => {
+      if (item.hasAttribute('href') && item.getAttribute('href'))
+        res.push(item.getAttribute('href') || '');
+      return res;
+    }, []);
+
+    paths.forEach((path) => {
+      expect(hrefs.includes(path)).toEqual(true);
+    });
+  });
+
+  it('redirects to the default page if the user is not authenticated and the page is protected', async () => {
+    initializeUser(false);
+    await act(async () => {
+      render(<Wrapper initial_entries={['/preferences']} />);
+    });
+    expect(location.assign).toHaveBeenCalledTimes(1);
+  });
+
+  it('does not redirect to the default page if the user is authenticated and the page is protected', async () => {
+    initializeUser(true);
+    await act(async () => {
+      render(<Wrapper initial_entries={['/preferences']} />);
+    });
+    expect(screen.getAllByRole('heading', { level: 2, name: 'My preferences' }).length).toEqual(1);
```

Comment:
You just have to write things like below :
```suggestion
    screen.getByRole('heading', { level: 2, name: 'My preferences' })
```

If no element is found, an error is thrown so the test fails